pkg/controller/statefulset/stateful_set_utils_test.go

```diff
@@ -80,11 +80,13 @@ func TestIdentityMatches(t *testing.T) {
 	}
 	pod = newStatefulSetPod(set, 1)
 	delete(pod.Annotations, podapi.PodHostnameAnnotation)
+	pod.Spec.Hostname = ""
```

Comment:
The field is set during `newStatefulSetPod`. This test is intentionally clearing the annotation/field to verify that causes the match to fail.